features/topology-map/plugins/org.opennms.features.topology.plugins.topo.graphml/src/main/java/org/opennms/features/topology/plugins/topo/graphml/GraphMLTopologyProvider.java

```diff
@@ -101,7 +111,30 @@ public GraphMLTopologyProvider(GraphMLGraph graph, GraphMLServiceAccessor servic
         focusStrategy = getFocusStrategy(graph);
         focusIds = getFocusIds(graph);
         preferredLayout = graph.getProperty(GraphMLProperties.PREFERRED_LAYOUT);
-        requiresStatusProvider = graph.getProperty(GraphMLProperties.VERTEX_STATUS_PROVIDER, Boolean.FALSE);
+
+        final Object vertexStatusProviderType = graph.<Object>getProperty(GraphMLProperties.VERTEX_STATUS_PROVIDER, Boolean.FALSE);
```

Comment:
Can we move this `vertexStatusProviderType` detection code in its own method, please.